src/main/kotlin/com/salesforce/dynamodbv2/mt/backups/MtBackupAwsAdaptor.kt

```diff
@@ -51,11 +51,39 @@ class MtBackupAwsAdaptor {
                 BackupDescription().withBackupDetails(getBackupDetails(mtBackupMetadata)))
     }
 
+    fun getDescribeBackupResult(backupMetadata: TenantBackupMetadata): DescribeBackupResult {
+        return DescribeBackupResult().withBackupDescription(
+                BackupDescription().withBackupDetails(getBackupDetails(backupMetadata))
+        )
+    }
+
+    fun getTenantTableBackupFromArn(backupArn: String): TenantTableBackupMetadata {
+        if (!isTenantTableArn(backupArn)) {
+            throw IllegalArgumentException("$backupArn does not include tenant-table specifier.")
+        }
+        val tenantTableParts = backupArn.split(':')
+        return TenantTableBackupMetadata(tenantTableParts[0], tenantTableParts[1], tenantTableParts[2])
+    }
+
+    private fun isTenantTableArn(backupArn: String): Boolean = backupArn.split(':').size == 3
+
+    fun getBackupArnForTenantTableBackup(tenantTable: TenantTableBackupMetadata): String =
+            "${tenantTable.backupName}:${tenantTable.tenantId}:${tenantTable.virtualTableName}"
```

Comment:
Have you considered using the same format we use for virtual stream arns, i.e., appending virtual table name and tenant id to the end of the arn?

For backup something like:
`arn:${Partition}:dynamodb:${Region}:${Account}:table/${TableName}/backup/${BackupName}/context/${tenantId}/tenantTable/${tenantTable}`

(source: https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazondynamodb.html#amazondynamodb-resources-for-iam-policies)

Reply:
will take a stab trying to align with that... I think it might make sense to use the tenant table backup metadata json file ARN for the s3 resource, so its actually a useful ARN vs arbitrarily aligned